File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> hs = new HashMap<>();
        
        for(int i = 0; i < nums.length ; i++){
            if( hs.containsKey(nums[i]) ){
                Integer itr = hs.get(nums[i]);
                if( i - itr <= k) {
                    return true;
                }
            }
            hs.put(nums[i] , i);
        }
        return false;
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.go
func containsNearbyDuplicate(nums []int, k int) bool {
    mp := map[int]int{}
    for idx, val := range nums {
        if index, found := mp[val]; found && idx - index <= k  {
            return true
        }
        mp[val] = idx
    }
    return false
}